### ISSUE TITLE
Support react 17

### ISSUE THREAD

**Issue by Kjagd** [JustFly1984/react-google-maps-api#1684]
Due to the peer dependency on react 16, it is currently not possible to use the library with react 17.

https://github.com/JustFly1984/react-google-maps-api/blob/develop/packages/react-google-maps-api/package.json#L92

```
"peerDependencies": {
    "react": "^16.6.3",
    "react-dom": "^16.6.3"
  },
```

## Your Environment

os: linux

node --version
15.1.0

react version
17.0.1

@react-google-maps/api version
1.13.0

**Comment by JustFly1984** [comment]
I've tried to update to react 17 and typescript 4.1-beta, but at some point we had our build tool changed from webpack to tsdx, and tsdx do not support beta typescript, and react 17, so I wait if they add the support. If somebody want to champion the rollback to webpack, you can look the github history for webpack config, and try to reimplement new webpack 5 build instead of tsdx.

I had multiple issues with tsdx for as long as it was added, and would love to go back to webpack, but have no time for transition right now.

**Comment by schmidtfx** [comment]
Any updates on this?

**Comment by JustFly1984** [comment]
if somebody want to make PR - you are welcome.

**Comment by JustFly1984** [comment]
released 2.0.1 with typescript 4.1.2 and react 17 support
